miden-lib/asm/kernels/transaction/lib/tx.masm

```diff
@@ -178,6 +184,41 @@ proc.add_non_fungible_asset_to_note
     # => [note_ptr, note_idx]
 end
 
+#! Sets the transaction expiration delta, in blocks.
+#! Once set, the delta can be later decreased but not increased. 
+#!
+#! The input block height delta added to the reference block in order to output an upper limit
+#! at which the transaction will be considered valid (not expired).
+#!
+#! Inputs: [block_height_delta, ...]
+#! Output: [...]
+#!
+#! Where:
+#! - block_height_delta is the desired expiration time delta (1 to 0xFFFF).
+#! - tx_block_height_delta is the updated (or unchanged) transaction expiration time delta.
+export.set_expiration_block_num
```

Comment:
I think the description (especially line 187) should be updated. Also, it may be good to rename this to something like `update_expiration_block_num` as we are not setting it directly.

Lastly, should we not return the "old delta" form here? Alternatively, and this is probably a better option, we could add `get_expiration_delta` procedure so that users could get it from the kernel at any point in time.

Reply:
Updated the name.
> I think the description (especially line 187) should be updated

In what sense where you thinking it should be updated? The user of the function would still be setting the expiration delta, no?

> Lastly, should we not return the "old delta" form here? Alternatively, and this is probably a better option, we could add get_expiration_delta procedure so that users could get it from the kernel at any point in time.

Initially I made it like this (also thought about returning just the stored delta), but I removed it in favor of a getter. I added the getter now.

Reply:
> In what sense where you thinking it should be updated? The user of the function would still be setting the expiration delta, no?

I think here we can say that we are updating transaction expiration block number since this is an internal procedure and this is more reflective of what the procedure actually does.